core-rust/state-manager/src/mempool/pending_transaction_result_cache.rs

```diff
@@ -21,6 +22,7 @@ pub enum RejectionReason {
     FromExecution(Box<RejectionError>),
     ValidationError(TransactionValidationError),
     IntentHashCommitted,
+    VertexLimit(VertexLimitsExceeded),
```

Comment:
It doesn't feel to me that this should be a rejection - rejections all currently imply "at some state version, this transaction was rejected / uncommittable". And we handle rejections by (eg) booting the transaction from the mempool and adding them to the rejection cache.

I realise you've given it a base delay of 0, but even so, it feels like it shouldn't be a rejection to start with.